Source/EasyNetQ/Topology/Queue.cs

```diff
@@ -1,68 +1,56 @@
 ﻿using System.Collections.Generic;
-using System.Linq;
+using System.Collections.ObjectModel;
 
 namespace EasyNetQ.Topology
 {
-    /// <inheritdoc />
-    public sealed class Queue : IQueue
+    /// <summary>
+    ///     Represents an AMQP queue
+    /// </summary>
+    public readonly struct Queue
     {
         /// <summary>
         ///     Creates Queue
         /// </summary>
-        public Queue(string name, bool isDurable = true, bool isExclusive = false, bool isAutoDelete = false,
-            IDictionary<string, object> arguments = null)
+        public Queue(
+            string name,
+            bool isDurable = true,
+            bool isExclusive = false,
+            bool isAutoDelete = false,
+            IDictionary<string, object> arguments = null
+        )
         {
             Preconditions.CheckNotBlank(name, "name");
 
             Name = name;
             IsDurable = isDurable;
             IsExclusive = isExclusive;
             IsAutoDelete = isAutoDelete;
-            Arguments = arguments ?? new Dictionary<string, object>();
+            Arguments = arguments == null ? null : new ReadOnlyDictionary<string, object>(arguments);
```

Comment:
It is a breaking change for existing code because of possible NRE. Suppose to return cached `ReadOnlyDictionary` from static field.

Reply:
agree